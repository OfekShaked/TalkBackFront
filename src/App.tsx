import React from 'react';
import logo from './logo.svg';
import {SocketContext, socket} from '../src/context/socketContext';
import './App.scss';
import PageLayout from './components/page-layout/PageLayout';

function App() {
  return (
    <div className="App">
      <SocketContext.Provider value={socket}>
      <PageLayout></PageLayout>
      </SocketContext.Provider>
    </div>
  );
}

export default App;
