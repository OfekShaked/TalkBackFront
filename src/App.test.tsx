import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import SocketMock from 'socket.io-mock';

test('renders app page', () => {
  const component = renderer.create(
    <App ></App>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Fast and isolated socket tests', ()=>{
  it('Sockets should be able to talk to each other without a server', ()=> {
      let socket = new SocketMock();

      socket.on('message',  (message:any) =>{
          expect(message).toBe('Hello World!');
      });
      socket.socketClient.emit('message', 'Hello World!');
  });
});
