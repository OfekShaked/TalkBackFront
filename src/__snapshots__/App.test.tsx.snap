// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app page 1`] = `
<div
  className="App"
>
  <div
    className="makeStyles-root-1"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-5 mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-7 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
        >
          What's On My App
        </h6>
      </div>
    </header>
    <div
      className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-9"
    >
      <div
        className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-10 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <div
          className="makeStyles-drawerHeader-11"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </span>
          </button>
        </div>
        <hr
          className="MuiDivider-root"
        />
        <ul
          className="MuiList-root MuiList-padding"
        >
          <a
            aria-current={null}
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            href="/signin"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Sign-In
              </span>
            </div>
          </a>
          <a
            aria-current={null}
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            href="/register"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Register
              </span>
            </div>
          </a>
        </ul>
      </div>
    </div>
    <main
      className="makeStyles-content-3"
    >
      <div
        className="makeStyles-drawerHeader-2"
      />
      <div
        className="vc-container"
      >
        <div
          className="vc-content"
        >
          <h1
            className="vc-heading"
          >
            500
          </h1>
          <p
            className="vc-sub-heading"
          >
            Internal Server 
            <span
              className="blink-infinite"
            >
              Error 
            </span>
          </p>
        </div>
      </div>
    </main>
  </div>
</div>
`;
